### DISCUSSION TITLE
IHostResources

### DISCUSSION THREAD

**Discussion by sbwalker** [oqtane/oqtane.framework#2818] [General]
The IHostResources interface was introduced way back in 2020 as a work-around to allow developers to add static resource references for CSS and JS libraries to the page head. The problem with this interface is that it does not support multi-tenancy - which basically means that if you use it, the resources specified will be included in every page in every site in your entire installation. In addition, it relies on reflection which means that every server-side request iterates through the assemblies in the app domain to determine if they implement the interface - which is not optimal for performance. I would like to deprecate this interface in Oqtane 4.0.0 (.NET 7) as I believe there are now other more mature options available in the framework which do not have the limitations explained above.  But I would first like to hear from developers who are using it to ensure I am not missing something.

**Comment by nicpitsch** [comment]
I use IHostResources to load Telerik Blazor js and css for Telerik components and Telerik reporting. The usage is based on your comment in [discussion#690](https://github.com/oqtane/oqtane.framework/discussions/690). 

corrected based on @sbwalker response: Provided that it is possible to load the resources in Oqtane 4 at Site ~~Tenant~~ level, I assume the change should not cause problems.

**Comment by sbwalker** [comment]
@nicpitsch yes, #690 is the thread from 2020 when the interface was introduced. I don't believe its a good idea to have resources defined tenant-wide - but you do need the ability to define resources at the site level.

**Comment by markdav-is** [comment]
We use IHostResources in a number of our apps, but knowing it will be deprecated, we could move things to themes or modules as needed.

**Comment by sbwalker** [comment]
@markdav-is IHostResources was deprecated in 4.0. The new approach is documented here: https://www.oqtane.org/blog/!/61/css-js-resources
